Feature: Persistency KVS -- Qorix Review

sbachmann-qorix opened the pull request.

sbachmann-qorix committed:
persistency_kvs: feature architecture

Signed-off-by: Sven Bachmann <sven.bachmann.ext@qorix.ai>

sbachmann-qorix committed:
persistency_kvs: feature architecture

Signed-off-by: Sven Bachmann <sven.bachmann.ext@qorix.ai>

sbachmann-qorix committed:
persistency_kvs: component requirements

Signed-off-by: Sven Bachmann <sven.bachmann.ext@qorix.ai>

vinodreddy-g commented on docs/features/persistency/kvs/requirements/index.rst:
```diff
@@ -0,0 +1,351 @@
+
+   Requirements must be linked to top-level (Stakeholder) requirements.
+
+.. feat_req:: Document requirements as code
```
We don't need this requirement at feature level.

sbachmann-qorix replied:
Removed

vinodreddy-g commented on docs/features/persistency/kvs/requirements/index.rst:
```diff
@@ -0,0 +1,351 @@
+
+   The KVS must use the provided S-CORE logging and tracing framework.
+
+.. feat_req:: Development steering
```
We don't need this requirement at feature level.

sbachmann-qorix replied:
Removed

vinodreddy-g commented on docs/features/persistency/kvs/requirements/index.rst:
```diff
@@ -0,0 +1,351 @@
+
+   The KVS must run on all S-CORE platforms.
+
+.. feat_req:: Logging and tracing
```
We don't need this requirement at feature level.

sbachmann-qorix replied:
Removed

vinodreddy-g commented on docs/features/persistency/kvs/requirements/index.rst:
```diff
@@ -0,0 +1,351 @@
+
+   The KVS must support concurrent access to the data.
+
+.. feat_req:: Multi-architecture support
```
We don't need this requirement at feature level.

sbachmann-qorix replied:
Removed

vinodreddy-g commented on docs/features/persistency/kvs/requirements/index.rst:
```diff
@@ -0,0 +1,351 @@
+   :satisfies: stkh_req__communication__intra_process
+   :status: valid
+
+   The KVS must support concurrent access to the data.
```
shall support

sbachmann-qorix replied:
Ok, fixed

vinodreddy-g commented on docs/features/persistency/kvs/requirements/index.rst:
```diff
@@ -0,0 +1,351 @@
+   :satisfies: stkh_req__execution_model__startup_perf
+   :status: valid
+
+   The KVS must ensure that the startup time, besides the time that is needed
```
This is not a testable requirement we need concrete numbers if any like less than 10ms or something.

sbachmann-qorix replied:
Removed

vinodreddy-g commented on docs/features/persistency/kvs/requirements/index.rst:
```diff
@@ -0,0 +1,351 @@
+   :satisfies: stkh_req__execution_model__short_app_cycles
+   :status: valid
+
+   The KVS must ensure that operations are usually done in under 5 ms.
```
We need to rewrite it as formal requirement.

sbachmann-qorix replied:
Done

vinodreddy-g commented on docs/features/persistency/kvs/requirements/index.rst:
```diff
@@ -0,0 +1,351 @@
+   :satisfies: stkh_req__app_architectures__support_data
+   :status: valid
+
+   The KVS must provide an API to register callbacks which will be called for
```
May be this  -
The KVS shall provide an API that allows for the registration of callback functions. These callbacks shall be invoked in response to specific events, such as when keys are changed or removed.

sbachmann-qorix replied:
Done

vinodreddy-g commented on docs/features/persistency/kvs/requirements/index.rst:
```diff
@@ -0,0 +1,351 @@
+   :satisfies: stkh_req__dependability__automotive_safety
+   :status: valid
+
+   The KVS must support at least ASIL-B.
```
This is not required here.

sbachmann-qorix replied:
Removed

vinodreddy-g reviewed (commented):
i think the requirements structuring and words need to checked in general .And the generic requirements which applies to all modules need not be at KVS level.

sbachmann-qorix committed:
persistency_kvs: synced components in feature architecture

Signed-off-by: Sven Bachmann <sven.bachmann.ext@qorix.ai>

sbachmann-qorix committed:
persistency_kvs: integrate review points from Volker

See https://github.com/eclipse-score/score/pull/1068

Signed-off-by: Sven Bachmann <sven.bachmann.ext@qorix.ai>

sbachmann-qorix committed:
persistency_kvs: integrate review points from Volker

See https://github.com/eclipse-score/score/pull/1068

Signed-off-by: Sven Bachmann <sven.bachmann.ext@qorix.ai>

vinodreddy-g committed:
docs: added module and corrected the interface in puml

sbachmann-qorix committed:
persistency_kvs: map all dynamic fulfilled feature reqs to static arch

Signed-off-by: Sven Bachmann <sven.bachmann.ext@qorix.ai>

sbachmann-qorix committed:
persistency_kvs: map all dynamic fulfilled feature reqs to static arch

Signed-off-by: Sven Bachmann <sven.bachmann.ext@qorix.ai>

sbachmann-qorix committed:
persistency_kvs: reword component req after IREB standard

Signed-off-by: Sven Bachmann <sven.bachmann.ext@qorix.ai>

sbachmann-qorix committed:
persistency_kvs: reword component reqs to match the template

https://eclipse-score.github.io/score/pr-1069/process/process_areas/requirements_engineering/guidance/requirements_templates.html#gd_temp__req__formulation

Signed-off-by: Sven Bachmann <sven.bachmann.ext@qorix.ai>

sbachmann-qorix committed:
persistency_kvs: integrated review comments from feature architecture

Signed-off-by: Sven Bachmann <sven.bachmann.ext@qorix.ai>

PandaeDo committed:
process: templates for feature and module persistency added

sbachmann-qorix committed:
persistency_kvs: move KVS reqs to new folder structure

Signed-off-by: Sven Bachmann <sven.bachmann.ext@qorix.ai>

sbachmann-qorix committed:
persistency_kvs: renamed kvs to persistency in key names

Signed-off-by: Sven Bachmann <sven.bachmann.ext@qorix.ai>

sbachmann-qorix committed:
persistency_kvs: move KVS arch to new folder structure

Signed-off-by: Sven Bachmann <sven.bachmann.ext@qorix.ai>

sbachmann-qorix committed:
persistency_kvs: integrated feature template

Note: This doesn't build completely and contains some id errors.

Signed-off-by: Sven Bachmann <sven.bachmann.ext@qorix.ai>

PandaeDo committed:
process: fix errors that documentation is buildable

PandaeDo committed:
process: fix build errors

sbachmann-qorix committed:
persistency_kvs: move KVS to new directory structure

Signed-off-by: Sven Bachmann <sven.bachmann.ext@qorix.ai>

sbachmann-qorix committed:
persistency_kvs: set feature status to valid

Signed-off-by: Sven Bachmann <sven.bachmann.ext@qorix.ai>

sbachmann-qorix committed:
persistency_kvs: adapt needs-links to externalized process docs

Signed-off-by: Sven Bachmann <sven.bachmann.ext@qorix.ai>

sbachmann-qorix committed:
persistency_kvs: removed feat arch & comp reqs for a clean PR

Signed-off-by: Sven Bachmann <sven.bachmann.ext@qorix.ai>

sbachmann-qorix committed:
persistency_kvs: re-add feature architecture

Signed-off-by: Sven Bachmann <sven.bachmann.ext@qorix.ai>

sbachmann-qorix committed:
persistency_kvs: re-add missing Persistency main topic to KVS

Signed-off-by: Sven Bachmann <sven.bachmann.ext@qorix.ai>
